server-side/src/auth/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { RegAuthDto } from './dto/reg-auth.dto';
+import { LoginAuthDto } from './dto/login-auth.dto';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+import { Response } from 'express';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectModel('user') private usersModel,
+    private jwt: JwtService,
+  ) {}
+
+  async Login(loginAuthDto: LoginAuthDto, res: Response) {
+    let foundUser = await this.usersModel.findOne({
+      email: loginAuthDto.email,
+    });
+    if (!foundUser) return { message: 'Invalid Email Or Password !!' };
+    let isTruePass = await bcrypt.compare(
+      loginAuthDto.password,
+      foundUser.password,
+    ); //true||false
+    if (!isTruePass) return { message: 'Invalid Email Or Password !!' };
+    let myJWT = await this.jwt.sign(
+      { name: foundUser.name },
+      { secret: '123' },
+    );
+    res.header('token', myJWT);
+
```

Comment:
- No need to pass Response object and set headers manually. Return the token directly to the client.
  - Use the user's ID in the token payload for better security.
 - Use NotFoundException for invalid email or password instead of returning a message.
      example :  if (!foundUser) throw new NotFoundException('Invalid Email Or Password !!' )